scripts/generate_config_tests.py

```diff
@@ -160,15 +159,10 @@ class ConfigTestGenerator(test_data_generation.TestGenerator):
     """Generate test cases for configuration reporting."""
 
     def __init__(self, settings):
-        # Temporarily use different config classes for 3.6. With the config.py moving to
-        # the framework it will be unified.
-        is_3_6 = not isinstance(config.ConfigFile, abc.ABCMeta)
-        # pylint: disable=no-value-for-parameter, no-member
-        self.mbedtls_config = config.ConfigFile() if is_3_6 else config.MbedTLSConfig()
+        self.mbedtls_config = config.MbedTLSConfig()
         self.targets['test_suite_config.mbedtls_boolean'] = \
             lambda: enumerate_boolean_setting_cases(self.mbedtls_config)
-        self.psa_config = config.ConfigFile('include/psa/crypto_config.h') if is_3_6 else \
-                          config.CryptoConfig()
+        self.psa_config = config.CryptoConfig()
```

Comment:
@gabor-mezei-arm
Now, in build_tree.py we have an is_mbedtls_3_6() method.
Is it the case that if we restore here something like:
```py
self.psa_config = config.ConfigFile('include/psa/crypto_config.h') if build_tree.is_mbedtls_3_6() else \
config.CryptoConfig.h
```
we could then remove CryptoConfig() and CryptoConfigFile() from config.py in mbedtls-3.6?

Not saying that we should do it, just checking my understanding.

Reply:
`ConfigFile` changed to an abstract class. It can not be instantiated.